lib/src/main/java/io/ably/lib/realtime/MessageExtrasFilter.java

```diff
@@ -0,0 +1,93 @@
+package io.ably.lib.realtime;
+
+import com.google.gson.JsonObject;
+import io.ably.lib.types.Message;
+
+/**
+ * A filter for inspecting the extras of a message to check if they match.
+ * Spec: RTL22c
+ */
+final class MessageExtrasFilter implements FilteredListeners.IMessageFilter {
+    public final io.ably.lib.types.MessageFilter filter;
+
+    MessageExtrasFilter(io.ably.lib.types.MessageFilter filter) {
+        this.filter = filter;
+    }
+
+    @Override
+    public boolean onMessage(Message message) {
+        return clientIdMatchesFilter(message) &&
+            nameMatchesFilter(message) &&
+            referenceMatchesFilter(message);
+    }
+
+    private boolean nameMatchesFilter(Message message) {
+        return filter.name == null || message.name.equals(filter.name);
```

Comment:
I mean it would make sense to have a powerful filtering capability like regex here and other places too : )